Source/DafnyCore/Resolver/PreType/PreTypeResolve.State.cs

```diff
@@ -0,0 +1,886 @@
+//-----------------------------------------------------------------------------
+//
+// Copyright by the contributors to the Dafny Project
+// SPDX-License-Identifier: MIT
+//
+//-----------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics.Contracts;
+using Microsoft.Boogie;
+
+namespace Microsoft.Dafny {
+  public partial class PreTypeResolver {
```

Comment:
Please remove the partial. We can add it back in a future PR is if turns out to be warranted there, but I expect it'll be better to pass `PreTypeResolver` around more.

The downside of having a lot of `partial class` declarations is that the fields of this class become spread of many files, and it becomes difficult to both see and limit the size of the class's state.